src/map.ts

```diff
@@ -0,0 +1,88 @@
+import 'leaflet/dist/leaflet.css';
+import L from "leaflet";
+import getWeatherData from './apiHelper.ts';
+import { CityInterface } from './interfaces';
+import { toggleNav } from './domManipulation.ts';
+
+export function initializeMap(): L.Map {
+    const map = L.map('map').setView([-30.5595, 22.9375], 5); //set view to South Africa
+
+    L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    maxZoom: 19,
+    attribution: '© OpenStreetMap',
+    }).addTo(map);
+
+    return map;
+}
+
+export function addPinsToMap(map: L.Map, cities: CityInterface[]): void {
+    cities.forEach((city) => {
+        const marker = L.marker([city.lat, city.lng], {
+          riseOnHover: true,
+          title: city.city_name,
+          autoPanOnFocus: true,
+        }).addTo(map);
+        marker.options.riseOnHover = true;
+      
+        const cityList = document.getElementById('city-list');
+        const a = document.createElement('a');
```

Comment:
Why did you opt for an anchor tag instead of a button if you don't want to use the `href` attribute?